### ISSUE TITLE
compilation of packages on Windows can fail if RTOOLS40_HOME unset

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#6746]
### System details

    RStudio Edition : Desktop
    RStudio Version : any
    OS Version      : Windows 10 x64
    R Version       : 4.0.0

### Steps to reproduce the problem

Use RStudio to install Rtools; e.g. via

```
if (!file.exists("C:/RBuildTools/4.0"))
  .rs.installBuildTools("Installing Rtools")
```

Then, put that on the PATH:

```
path <- Sys.getenv("PATH")
Sys.setenv(PATH = paste("C:\\RBuildTools\\4.0\\usr\\bin", path, sep = ";"))
```

Then, ensure that `RTOOLS40_HOME` is unset; e.g.

```
Sys.unsetenv("RTOOLS40_HOME")
```

Finally, try to install a package from sources that requires compilation, e.g.

```
install.packages("digest", type = "source")
```

### Describe the problem in detail

Installation fails as the wrong path to Rtools40 is used:

```
> install.packages("digest", type = "source")
Installing package into 'C:/Users/kevinushey/r/win-library/4.0'
(as 'lib' is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/digest_0.6.25.tar.gz'
Content length 145642 bytes (142 KB)
downloaded 142 KB

* installing *source* package 'digest' ...
** package 'digest' successfully unpacked and MD5 sums checked
** using staged installation
** libs

*** arch - i386
"c:/rtools40/mingw32/bin/"g++  -std=gnu++11 -I"C:/R/R-40~1.0/include" -DNDEBUG     -I"C:/Users/kevinushey/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign -c SpookyV2.cpp -o SpookyV2.o
sh: c:/rtools40/mingw32/bin/g++: No such file or directory
make: *** [C:/R/R-40~1.0/etc/i386/Makeconf:229: SpookyV2.o] Error 127
ERROR: compilation failed for package 'digest'
* removing 'C:/Users/kevinushey/r/win-library/4.0/digest'
* restoring previous 'C:/Users/kevinushey/r/win-library/4.0/digest'
Warning in install.packages :
  installation of package 'digest' had non-zero exit status

The downloaded source packages are in
	'C:\tmp\RtmpemEXb7\downloaded_packages'
```

### Describe the behavior you expected

The correct path to the compiler tools should be used.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by R-Hannibal** [comment]
@kevinushey - I'm having trouble verifying this one. Probably an issue with my environment or something, but when running the reproduction steps on either 1.2 or the latest 1.3 preview I am getting the same result after running the final command (`install.packages("digest", type = "source")`) in the console:

Seems to be an endless loop with the 'Updating Loaded Packages' prompt, do you have any ideas what could be causing this?

![UpdatingLoadedPackagesLoop](https://user-images.githubusercontent.com/62304541/85902851-92a84700-b7ca-11ea-82de-8e0d138e1f5b.gif)

**Comment by kevinushey** [comment]
Usually, this is caused by something that is automatically loading the `digest` package when the session is restarted. Some things worth checking:

1. If you clear your R environment with `rm(list = ls())`, and then try to install things, does it work?
2. Do you have anything in your `~/.Rprofile` (or similar) that automatically loads R packages on startup?

**Comment by R-Hannibal** [comment]
I'm still having trouble here (although now that I've switched to my new laptop which is a much 'cleaner' environment the error I'm getting is much more similar to what was reported/seems more legitimate). Again this occurs on the final command `install.packages("digest", type = "source")` on both 1.3.959 and 1.3.1054 - 

 ```
* installing *source* package 'digest' ...
** package 'digest' successfully unpacked and MD5 sums checked
** using staged installation
** libs

*** arch - i386
"c:/rtools40/mingw32/bin/"g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-40~1.2/include" -DNDEBUG          -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign -c SpookyV2.cpp -o SpookyV2.o
sh: c:/rtools40/mingw32/bin/g++: No such file or directory
make: *** [C:/PROGRA~1/R/R-40~1.2/etc/i386/Makeconf:229: SpookyV2.o] Error 127
ERROR: compilation failed for package 'digest'
* removing 'C:/Users/iredh/Documents/R/win-library/4.0/digest'
* restoring previous 'C:/Users/iredh/Documents/R/win-library/4.0/digest'
Warning in install.packages :
  installation of package ‘digest’ had non-zero exit status
```

Tried running `rm(list = ls())` and then re-running `install.packages("digest", type = "source")` , same result.

**Comment by gtritchie** [comment]
Interesting, I've only ever installed RTools on Windows as a separate downloadable install, didn't know about doing it via   .rs.installBuildTools!

https://cran.r-project.org/bin/windows/Rtools/

Those instructions have you set RTOOLS40_HOME in .Renviron file after installing.

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.3.1054, R Version - 4.0.2, Windows 10 - Desktop x64 

Comparison tested successfully between 1.3.959 vs. 1.3.1054, was able to reproduce the same error on 1.3.959. Same steps on 1.3.1054 installed the digest package from source successfully, no errors.
